### PR TITLE
enhance(chat): add scoped Doc Query runtime

### PR DESCRIPTION
**Pull request by rschlaefli**
## What This Adds

This is layer 1 of the three-PR replacement for #5709. It adds only the Chat runtime needed to call the shared multi-tenant Doc Query service with a knowledge-base scope.

- Resolve one enabled `KB` configuration for the effective chatbot mode, including the existing Tutor-to-Quizzer inheritance.
- Mint a five-minute ES256 scope token containing the knowledge-base, chatbot, thread, and request identities.
- Keep that scope token separate from the existing transport bearer in `X-Doc-Query-Scope-Token`.
- Fail closed before provider or message work when the required binding or scope configuration is missing, malformed, duplicated, or conflicting.

Chatbots without an enabled `KB` binding and non-`KB` MCP servers retain their existing behavior. This PR does not activate any chatbot or access production.

## Stack Position

- Base: `v3`
- This PR: scoped Doc Query runtime
- Next: #5737 adds the separately invoked activation operator
- Top: #5738 adds the separately invoked cutover proof
- Replaces the runtime portion of #5709; that source PR remains unchanged during stack validation.

## Branch Coverage

- Head: `732f7b3ef6155139565b596fe63f08d05a360666`
- Reviewed: 4 commits; 813 lines added and 11 removed across 10 substantive source, test, configuration, ADR, and documentation files, excluding the committed project plan.
- Covered: scope-token signer, binding resolver, MCP transport wiring, route fail-closed behavior, Tutor-to-Quizzer inheritance, Turborepo environment declarations, ADR 0021 alignment, and Chat platform documentation.

## Review Focus

- Confirm the effective selected binding must match the globally validated server and `kb_id`.
- Confirm the five-minute scope token never replaces or exposes the transport bearer.
- Confirm unscoped chatbots retain their prior behavior while malformed required scope fails before provider or message work.

## Verification

Current head:

- Focused Chat suites: 29 tests passed.
- Chat TypeScript check: passed.
- Focused ESLint, repository formatting, and `git diff --check`: passed.
- Commit and push hooks: passed; 23 workspace build tasks completed during publication.
- Staged Gitleaks scan: no leaks found.
- Dedicated simplification and security correction reviews: passed.
- GitHub exact-head hosted Playwright: build plus all 8 shards and the aggregate passed.
- GitHub security, CodeQL, GraphQL, fallback build, and OpenCodeReview checks passed.
- GitHub `check` is red only because current `v3@afba912051` contains two Biome-unformatted evaluation files outside this PR's diff: `apps/chat/scripts/klicker-evaluation-target.mjs` and `apps/chat/test/klicker-evaluation-target.test.mjs`.
- The repository final AI review remains intentionally untriggered while this PR is draft.

Warning:

- The local host used Node 26 while the repository pins Node 24. Hosted CI remains the runtime-version gate.
- `origin/v3` advanced by two commits after the branch was reviewed. The only shared path is `docs/chat-platform.md`, changed in separate hunks; GitHub reports this draft mergeable.

## Security / Privacy

- No bearer, signing key, secret value, production content, or personal data is committed.
- Signing inputs remain environment references, and the scope token is attached only to the scoped Doc Query request.

## Blocking Before Merge

- [ ] The target-branch-only formatting failure is corrected or an approved upstream integration brings its correction into this branch.
- [ ] Trigger and pass the repository final AI review after separate ready-for-review approval.
- [ ] Review the five-minute token lifetime during production deployment preflight.
- [ ] Merge through the stack in dependency order; this plan does not authorize merge.

## Follow-Up After Merge

- [ ] Review and merge #5737 and #5738 separately.
- [ ] Keep activation, deployment, secret access, production writes, and live proof behind their existing external-action gates.

## Local Session Artifacts

- Machine: local macOS host
- Worktree: `trees/doc-query-v3-scope-activation` in repository `klicker-uzh`

## Summary by CodeRabbit

* **New Features**
  * Added scoped knowledge-base retrieval for chatbot conversations.
  * Chat requests now securely authorize access to the selected knowledge base and conversation.
  * Quizzer mode can inherit eligible Tutor knowledge-base settings.

* **Bug Fixes**
  * Invalid, conflicting, or incomplete knowledge-base configurations now fail safely with a clear service-unavailable response.
  * Preserved existing behavior for chatbots without knowledge-base integrations and for other MCP services.

* **Documentation**
  * Documented knowledge-base binding requirements, inheritance rules, authorization, and configuration guidelines.

### CODE REVIEW

File: apps/chat/src/services/mcpClients.ts [hunk 1]
```diff
@@ -105,17 +113,65 @@ function toSafeToolName(
...
+  }
+  if (
+    typeof options.sessionId !== 'string' ||
+    options.sessionId.trim().length === 0
+  ) {
+    throw new Error('Scoped knowledge retrieval is not available')
+  }
+
+  const kbId = normalizeDocQueryKbId(options.kbId)
+  headers.Authorization = `Bearer ${safeDecrypt(server.authSecret)}`
```
[the remark this answers is not in the document]
Comment (rschlaefli) [reply apps/chat/src/services/mcpClients.ts:141]: Fixed in 46e8a4b7aa: `assertDocQueryTransportSecurity` now runs before any credential is attached.

- HTTPS is required for every Doc Query endpoint; plain HTTP is accepted only for clearly internal targets (loopback, `*.svc`/`*.internal`/`*.local`, RFC1918 ranges) so in-cluster cleartext keeps working.
- Violations fail closed before `Authorization`/scope headers are constructed and surface as `REQUIRED_MCP_UNAVAILABLE`.
- Covered by focused tests: public-cleartext rejection, HTTPS/internal acceptance, and 172.16/12 boundary cases.
